sotodlib/io/bookbinder.py

```diff
@@ -183,54 +243,35 @@ def rebundle(self, flush_time):
 
         Returns
         -------
-        signalout : G3SuperTimestream or None
+        signalout : G3SuperTimestream
             Output signal timestream
         biasout : G3SuperTimestream or None
-            Output TES bias timestream
+            Output TES biases timestream
         primout : G3SuperTimestream or None
             Output primary timestream
         """
 
-        def rebundle_sts(sts, flush_time, use_rids=False):
-            """
-            Since G3SuperTimestreams cannot be shortened, the data must be copied to
-            a new instance for rebundling of the buffer
+        # Timestamps to be output and kept in buffer, respectively
+        # (assumes signal, bias, and primary have same timestamps, which is a reasonable assumption)
+        tout       = [t for t in self.times if t < flush_time]
+        self.times = [t for t in self.times if t >= flush_time]
 
-            Parameters
-            ----------
-            flush_time : G3Time
-                Output will contain buffered data up to (but not including) this time
-            use_rids : bool
-                Whether the `.names` field should be replaced with readout_ids
+        # Since G3SuperTimestreams cannot be shortened, the data must be copied to
+        # a new instance for rebundling of the buffer
+        signalout = so3g.G3SuperTimestream(self.signal.names, tout, self.signal.data[:,:len(tout)])
```

Comment:
I think it's always good to state our assumptions in terms of assertion statements. For example, are we assuming `len(tout) > 0` here?

Reply:
I don’t think so (but not 100% certain). G3SuperTimestream can be initialized with 0-length vectors, though I don’t know if it breaks anything down the line (in `FrameProcessor.flush`). Even more reason to have comprehensive tests!